### DISCUSSION TITLE
Unable to merge MutableHeaders with dict

### DISCUSSION THREAD

**Discussion by manlix** [encode/starlette#1515] [Potential Issue] [answered]
### Checklist

- [x] The bug is reproducible against the latest release and/or `master`.
- [x] There are no similar issues or pull requests to fix it yet.

### Describe the bug

Unable to merge `MutableHeaders` with `dict`.

### To reproduce

```python3
from starlette.datastructures import MutableHeaders

response = MutableHeaders()
response |= {"X-Header": "string"}
```
### Expected behavior

```python3
>>> response
MutableHeaders({'x-header': 'string'})
```

### Actual behavior

Raised exception:
```
TypeError: unsupported operand type(s) for |=: 'MutableHeaders' and 'dict'
```

### Environment

- OS: Ubuntu 21.04
- Python version: 3.9.6
- Starlette version: 0.15.0

### Additional context

Method `MutableHeaders.update()` can resolve the issue but I think comfortable is used to operators `|` (for `merge`) and `|=` (for `update`).

It's comfortable if you have many headers like:

```python3
...

headers = {
    "X-Header-1": "string-1",
    "X-Header-1": "string-2",
    "X-Header-3": "string-3",
    "X-Header-...": "string-...",
    "X-Header-N": "string-N",
}

response.headers |= headers  # equals to: response.headers.update(headers)

return response

...
```
It's like [Dictionary Merge & Update Operators in Python 3.9](https://docs.python.org/3/whatsnew/3.9.html#dictionary-merge-update-operators).

What do you think about it? Thank you.

**Comment by graingert** [comment]
imho this is a cPython bug. _collections_abc.Mapping should define these mixin methods

**Comment by Kludex** [comment, marked as answer]
Based on [PEP-584](https://www.python.org/dev/peps/pep-0584/#what-about-mapping-and-mutablemapping), it does look like a CPython bug :thinking: 

EDIT: I've read it again, and is not a bug on CPython. It's as intended.
